Platform/ApollolakeBoardPkg/Library/Stage2BoardInitLib/Stage2BoardInitLib.c

```diff
@@ -1294,12 +1294,24 @@ UpdateLoaderPlatformInfo (
 
   PlatformData  = (PLATFORM_DATA *)GetPlatformDataPtr ();
   if(PlatformData != NULL) {
-    LoaderPlatformInfo->SecurityFlags = (PlatformData->BtGuardInfo.Bpm.Vb & FeaturePcdGet(PcdVerifiedBootEnabled)) |
-                                        ((PlatformData->BtGuardInfo.Bpm.Mb & FeaturePcdGet(PcdMeasuredBootEnabled) & MEASURED_BOOT_ENABLED() ) << 1);
+    LoaderPlatformInfo->HwState = (PlatformData->BtGuardInfo.Bpm.Vb) |
+                                        ((PlatformData->BtGuardInfo.Bpm.Mb) << 1);
 
     // Get Manufacturing Mode from Heci
     ReadHeciFwStatus(&HeciFwSts);
-    LoaderPlatformInfo->SecurityFlags |= (HeciFwSts & BIT4) >> 2;
+    LoaderPlatformInfo->HwState |= (HeciFwSts & BIT4) >> 2;
+
+    if ((PlatformData->BtGuardInfo.Bpm.Vb != 0) && FeaturePcdGet(PcdVerifiedBootEnabled)) {
+      LoaderPlatformInfo->LdrFeatures |= FEATURE_VERIFIED_BOOT;
+    } else {
+      LoaderPlatformInfo->LdrFeatures &= ~FEATURE_VERIFIED_BOOT;
+    }
+
+    if ((PlatformData->BtGuardInfo.Bpm.Mb != 0) && FeaturePcdGet(PcdMeasuredBootEnabled) && MEASURED_BOOT_ENABLED() ){
+      LoaderPlatformInfo->LdrFeatures |= FEATURE_MEASURED_BOOT;
+    } else {
+      LoaderPlatformInfo->LdrFeatures &= ~FEATURE_MEASURED_BOOT;
+    }
```

Comment:
we should init LdrFeatures  at stage1B PlatformFeaturesInit() so that other code could use this.
and we should use config data when updating LdrFeatures.
 if ( (PlatformData->BtGuardInfo.Bpm.Mb != 0)  && FeaturePcdGet(PcdMeasuredBootEnabled) &&  (FeaturesCfgData->Features.MeasuredBoot != 0) {
LoaderPlatformInfo->LdrFeatures |= FEATURE_MEASURED_BOOT;
} else {
  LoaderPlatformInfo->LdrFeatures &= ~FEATURE_MEASURED_BOOT;
}

Reply:
Sure, I would update LdrFeatures flags in PlatformFeaturesInit based on BTG and Cfg data.